core/chain_makers.go

```diff
@@ -246,6 +246,22 @@ func makeHeader(chain consensus.ChainReader, parent *types.Block, state *state.S
 		time = new(big.Int).Add(parent.Time(), big.NewInt(10)) // block time is fixed at 10 seconds
 	}
 
+	extra := types.ExtraFields_v2{
```

Comment:
hmm... mind if i ask why this change is needed in this PR? what's the context?

Reply:
because we start v2 block format at block 0 now. And some tests are using this `makeHeader` function to create a new block. And our block consensus logic require block header have V2 Extra field. So we need to change here.